internal/config/config.go

```diff
@@ -0,0 +1,85 @@
+package config
+
+import (
+	"flag"
+	"fmt"
+	"github.com/caarlos0/env/v6"
+	"github.com/joho/godotenv"
+	"net"
+	"os"
+	"regexp"
+)
+
+// Config - структура конфигурации приложения
+type Config struct {
+	Addr     string `env:"SERVER_ADDRESS"`    // Адрес сервера
+	BaseURL  string `env:"BASE_URL"`          // Базовый адрес результирующего сокращенного URL
+	FilePath string `env:"FILE_STORAGE_PATH"` // Путь к файлу для сохранения данных
+}
+
+// Default - функция для создания новой конфигурации с значениями по умолчанию
+func Default() *Config {
+	return &Config{
+		Addr:     "localhost:8080",
+		BaseURL:  "http://localhost:8080",
+		FilePath: "",
+	}
+}
+
+func OnFlag() *Config {
+	defaultValue := Default()
+	// Инициализация флагов командной строки
+	addr := flag.String("a", defaultValue.Addr, "HTTP-адрес сервера")
```

Comment:
```suggestion
        config := Config{}
	flag.StringVar(&config.Addr, "a", defaultValue.Addr, "HTTP-адрес сервера")
```
тут тоже можно сразу в структуру записывать значения